### PR TITLE
[LI-HOTFIX] Refactor the AlterIsrManager and TransferLeaderManager to avoid duplicate code

### PR DESCRIPTION
**Pull request by gitlw**
TICKET = N/A
LI_DESCRIPTION =
Currently, the AlterIsrManager and the TransferLeaderManager have some code duplications.
This PR refactored the duplicate logic into a common BrokerToControllerRequestManager.

EXIT_CRITERIA = When this change is accepted upstream and pulled to this repo.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/AlterIsrManager.scala [hunk 1]
```diff
@@ -78,10 +83,11 @@ object AlterIsrManager {
       time = time,
       metrics = metrics,
       config = config,
-      channelName = "alterIsr",
+      channelName = "AlterIsr",
```
Comment (lmr3796) [inline core/src/main/scala/kafka/server/AlterIsrManager.scala]: If no specific benefit, I'd recommend sticking with the original upstream implementation.

File: core/src/main/scala/kafka/server/TransferLeaderManager.scala [hunk 2]
```diff
@@ -19,56 +19,45 @@ package kafka.server
...
+    brokerId: Int): TransferLeaderManager = {
     val nodeProvider = MetadataCacheControllerNodeProvider(config, metadataCache)
 
     val channelManager = BrokerToControllerChannelManager(
       controllerNodeProvider = nodeProvider,
       time = time,
       metrics = metrics,
       config = config,
-      channelName = "transferLeader",
+      channelName = "TransferLeader",
```
Comment (lmr3796) [inline core/src/main/scala/kafka/server/TransferLeaderManager.scala]: nit as above

File: core/src/main/scala/kafka/server/BrokerToControllerRequestManager.scala [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+import org.apache.kafka.common.protocol.Errors
+import org.apache.kafka.common.requests.AbstractRequest
+import org.apache.kafka.common.utils.Time
+
+import java.util
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.{BlockingQueue, ConcurrentHashMap, LinkedBlockingQueue, TimeUnit}
+import scala.collection.mutable.ListBuffer
+
+trait RequestItem {
```
Comment (lmr3796) [inline core/src/main/scala/kafka/server/BrokerToControllerRequestManager.scala]: nit: maybe use a more specific `BrokerToControllerRequestItem` ?

**Review by lmr3796 — CHANGES_REQUESTED** [review]

File: core/src/main/scala/kafka/server/AlterIsrManager.scala (hunk 1 above)
Comment (gitlw) [reply core/src/main/scala/kafka/server/AlterIsrManager.scala]: Fixed

File: core/src/main/scala/kafka/server/TransferLeaderManager.scala (hunk 2 above)
Comment (gitlw) [reply core/src/main/scala/kafka/server/TransferLeaderManager.scala]: Fixed

File: core/src/main/scala/kafka/server/BrokerToControllerRequestManager.scala (hunk 3 above)
Comment (gitlw) [reply core/src/main/scala/kafka/server/BrokerToControllerRequestManager.scala]: Thanks, that makes sense.

**Review by lmr3796 — APPROVED** [review]

**Pull request merged**